backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/ActivitiesManager.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hyperledger.bpa.impl;
+
+import com.github.jknack.handlebars.internal.lang3.StringUtils;
+import io.micronaut.core.annotation.Nullable;
+import org.hyperledger.aries.api.connection.ConnectionState;
+import org.hyperledger.aries.api.issue_credential_v1.CredentialExchangeState;
+import org.hyperledger.aries.api.present_proof.PresentationExchangeRole;
+import org.hyperledger.aries.api.present_proof.PresentationExchangeState;
+import org.hyperledger.bpa.controller.api.activity.*;
+import org.hyperledger.bpa.model.Partner;
+import org.hyperledger.bpa.model.PartnerProof;
+import org.hyperledger.bpa.repository.BPACredentialExchangeRepository;
+import org.hyperledger.bpa.repository.PartnerProofRepository;
+import org.hyperledger.bpa.repository.PartnerRepository;
+
+import javax.inject.Inject;
+import java.util.*;
+
+public class ActivitiesManager {
+
+    @Inject
+    PartnerRepository partnerRepo;
+
+    @Inject
+    PartnerProofRepository proofRepository;
+
+    @Inject
+    BPACredentialExchangeRepository credExRepository;
+
+    static List<ConnectionState> connectionStates(ConnectionState... states) {
+        List<ConnectionState> results = new ArrayList<>();
+        for (ConnectionState state : states)
+            results.add(state);
+        return results;
+    }
+
+    static List<ConnectionState> connectionStatesForActivities() {
+        return connectionStates(ConnectionState.REQUEST, ConnectionState.INVITATION, ConnectionState.ACTIVE,
+                ConnectionState.INVITATION.RESPONSE);
+    }
+
+    static List<ConnectionState> connectionStatesForTasks() {
```

Comment:
I added a check for whether the flags are set or not. I think this only applies to tasks (no need to check for a to-do if the flag is auto), but it will still show in the ActivityLog.